src/js/components/DataTable/DataTable.js

```diff
@@ -191,7 +191,12 @@ const DataTable = ({
     }
   }, [select, setSelectedDataContext]);
 
-  const [rowExpand, setRowExpand] = useState([]);
+  const [rowExpand, setRowExpand] = useState(rowDetails?.expand || []);
+  useEffect(() => {
```

Comment:
Does this pattern introduce an additional render? Will default state value be set and then also call the useEffect which will set the state value?

Reply:
Verified...same number of renders unless caller changes rowDetails.expand